iox_query_influxql/src/plan/planner_rewrite_expression.rs

```diff
@@ -121,16 +121,218 @@
 //! [`Reduce`]: https://github.com/influxdata/influxql/blob/1ba470371ec093d57a726b143fe6ccbacf1b452b/ast.go#L4850-L4852
 //! [`EvalBool`]: https://github.com/influxdata/influxql/blob/1ba470371ec093d57a726b143fe6ccbacf1b452b/ast.go#L4181-L4183
 //! [`Eval`]: https://github.com/influxdata/influxql/blob/1ba470371ec093d57a726b143fe6ccbacf1b452b/ast.go#L4137
+use crate::plan::error;
 use crate::plan::util::Schemas;
 use arrow::datatypes::DataType;
-use datafusion::common::tree_node::{Transformed, TreeNode};
-use datafusion::common::{Result, ScalarValue};
+use datafusion::common::tree_node::{
+    Transformed, TreeNode, TreeNodeRewriter, TreeNodeVisitor, VisitRecursion,
+};
+use datafusion::common::{Column, Result, ScalarValue};
 use datafusion::logical_expr::{
     binary_expr, cast, coalesce, lit, BinaryExpr, Expr, ExprSchemable, Operator,
 };
+use datafusion::optimizer::utils::{conjunction, disjunction};
+use schema::{InfluxColumnType, InfluxFieldType};
+
+/// Perform a series of passes to rewrite `expr` in compliance with InfluxQL behavior
+/// in an effort to ensure the query executes without error.
+pub(super) fn rewrite_conditional_expr(expr: Expr, schemas: &Schemas) -> Result<Expr> {
+    let expr = expr.rewrite(&mut FixRegularExpressions { schemas })?;
+    let expr = rewrite_time_range_exprs(expr)?;
+    rewrite_conditional(expr, schemas)
+}
+
+/// Perform a series of passes to rewrite `expr`, used as a column projection,
+/// to match the behavior of InfluxQL.
+pub(super) fn rewrite_field_expr(expr: Expr, schemas: &Schemas) -> Result<Expr> {
+    rewrite_expr(expr, schemas)
+}
+
+/// Traverse `expr` and promote time range expressions to the root
+/// binary node on the left hand side and combined using the conjunction (`AND`)
+/// operator to ensure the time range is applied to the entire predicate.
+///
+/// Additionally, multiple `time = <timestamp>` expressions are combined
+/// using the disjunction (OR) operator, whereas, InfluxQL only supports
+/// a single `time = <timestamp>` expression.
+///
+/// # NOTE
+///
+/// Combining relational operators like `time > now() - 5s` and equality
+/// operators like `time = <timestamp>` with a disjunction (`OR`)
+/// will evaluate to false, like InfluxQL.
+fn rewrite_time_range_exprs(expr: Expr) -> Result<Expr> {
+    // Perform a quick check to see if there are any OR conditions,
+    // as if there aren't, we don't need to rewrite the expression.
+    let mut has_or = false;
+    expr.apply(&mut |expr| {
+        if matches!(
+            expr,
+            Expr::BinaryExpr(BinaryExpr {
+                op: Operator::Or,
+                ..
+            })
+        ) {
+            has_or = true;
+            Ok(VisitRecursion::Stop)
+        } else {
+            Ok(VisitRecursion::Continue)
+        }
+    })?;
+
+    if !has_or {
+        return Ok(expr);
+    }
+
+    let mut rw = SeparateTimeRanges {
+        ..Default::default()
+    };
```

Comment:
This type derives `Default` so seems like you could just do
```suggestion
    let mut rw = SeparateTimeRanges::default();
```